pkg/scheduler/scheduling_application.go

```diff
@@ -418,6 +420,15 @@ func (sa *SchedulingApplication) tryAllocate(headRoom *resources.Resource, ctx *
 	for _, request := range sa.sortedRequests {
 		// resource must fit in headroom otherwise skip the request
 		if !resources.FitIn(headRoom, request.AllocatedResource) {
+			eventCache := events.GetEventCache()
+			if eventCache.IsStarted() {
+				message := fmt.Sprintf("Application %s does not fit into %s queue", request.ApplicationID, request.QueueName)
+				event, err := createInsufficientQueueResourcesEvent(request.AskProto, message)
+				if err != nil {
+					log.Logger().Warn("Could not emit event")
+				}
+				eventCache.AddEvent(event)
```

Comment:
call `events.CreateRequestEventRecord()` directly: ask cannot be nil and ID is always set at this point.
```
if event, err := events.CreateRequestEventRecord(); err == nil {
	log.Logger().Warn("Event creation failed",
		zap.String("event message", message),
		zap.Error(err))
} else {
	eventCache.AddEvent(event)
}
```

Reply:
Fixed